backend/lib/endpoints/posts.js

```diff
@@ -30,21 +30,25 @@ async function routes(app) {
   app.get(
     "/",
     {
-      preValidation: [app.authenticate],
       schema: getPostsSchema,
     },
     async (req) => {
       const { userId } = req.query;
-      const [userErr, user] = await app.to(User.findById(userId));
-      if (userErr) {
-        throw app.httpErrors.notFound();
+      let user, userErr;
+      if (userId != null) {
```

Comment:
Thanks @Naraujo13. This looks like it resolves something I was going to ask about so happy to get it merged. I can't really test the logged-in flow because we need the user V2 with location etc which you're working on.

Only nit-picky suggestion for now I would make is remove `!= null` as seeing this I think we're expecting to maybe see a `null` when really it will be `undefined` or a string (`objectId`). 
```suggestion
      if (userId) {
```

Two suggestions for improvement related to this - for **post-MVP** refactoring backlog - would be to either update `app.authenticate` with `optional` arg, or add `app.optionallyAuthenticate` middleware for such a case instead of passing userId as query string arg. Additionally, also do the fetching of the user in the middleware and add to `req` so we're accessing `req.user` in all the endpoints using such middleware.

Reply:
Yeah good suggestion @joshmorel 
That will work here, just have to be careful in some cases where the `if ( something )` could be any truthy value but it would also fail for anything falsy such as `0 ` and `false` and `""`